src/sentry/snuba/discover.py

```diff
@@ -151,8 +151,13 @@ def transform_data(result, translated_columns, snuba_filter):
     def get_row(row):
         transformed = {}
         for key, value in row.items():
-            if isinstance(value, float) and math.isnan(value):
-                value = 0
+            if isinstance(value, float):
+                # 0 for nan, and none for inf were chosen arbitrarily, nan and inf are invalid json
+                # so needed to pick something valid to use instead
+                if math.isnan(value):
+                    value = 0
+                elif math.isinf(value):
+                    value = None
```

Comment:
This doesn't seem ideal since `None` is usually used for the absence of a value but I don't have a better suggestion here. We can try this and see what the feedback is.

Reply:
Yeah my other thought was the literal string `inf` but we'd have to handle that in cell actions i think? So lets revisit based on feedback